appsdk/configurable_test.go

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appsdk
+
+import (
+	"testing"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConfigurableDeviceNameFilter(t *testing.T) {
+	configurable := AppFunctionsSDKConfigurable{
+		Sdk: &AppFunctionsSDK{
+			LoggingClient: lc,
+		},
+	}
+
+	params := make(map[string]string)
+
+	// DeviceNames not in params
+	params[""] = ""
+	trx := configurable.DeviceNameFilter(params)
+	assert.Nil(t, trx, "return result from DeviceNameFilter should be nil")
+
+	// set DeviceNames but empty
+	params["DeviceNames"] = ""
+	trx = configurable.DeviceNameFilter(params)
+	assert.NotNil(t, trx, "return result from DeviceNameFilter should not be nil")
+
+	// set DeviceNames in params
+	deviceIDs := "GS1-AC-Drive01, GS1-AC-Drive02, GS1-AC-Drive03"
+	params["DeviceNames"] = deviceIDs
+
+	trx = configurable.DeviceNameFilter(params)
+	assert.NotNil(t, trx, "return result from DeviceNameFilter should not be nil")
```

Comment:
Could convert this to table-driven tests. Example 
https://github.com/edgexfoundry/go-mod-core-contracts/blob/ab099800f5bf055fb95c35604f41f5c22f3c7434/models/deviceprofile_test.go#L89

Reply:
So I changed it to have the loop cuz I wasn't sure how i felt about it and was curious about what it would look like. But, after doing it, and sending it to a couple folks on the team, both had the same comment that readability went down for the sake of compactness. I think this makes sense if we have a bunch of different values to run through the same function, which maybe we should do to achieve that value that it brings, but otherwise I think I'd rather break these up into individual tests such that they could be run in parallel as readability is crucial to our team.  But like I said, I've converted it and I will leave it.